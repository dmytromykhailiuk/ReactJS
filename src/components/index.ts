export * from "./Component";
export * from "./PureComponent";
export * from "./FunctionalComponent";
