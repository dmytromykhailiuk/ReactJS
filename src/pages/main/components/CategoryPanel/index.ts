export { default as CategoryPanel } from './CategoryPanel';
