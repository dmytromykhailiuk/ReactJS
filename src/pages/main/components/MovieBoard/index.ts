export { default as MovieBoard } from "./MovieBoard";
export { default as MovieBoardContainer } from "./MovieBoardContainer";
