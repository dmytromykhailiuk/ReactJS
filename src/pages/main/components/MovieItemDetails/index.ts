export { default as MovieItemDetails } from "./MovieItemDetails";
