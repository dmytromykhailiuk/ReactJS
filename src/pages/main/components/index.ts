export * from "./Banner";
export * from "./MovieBoard";
export * from "./SearchPanel";
export * from "./CategoryPanel";
export * from "./SortPanel";
export * from "./MovieList";
export * from "./MovieItem";
export * from "./MovieMenu";
export * from "./MovieModal";
