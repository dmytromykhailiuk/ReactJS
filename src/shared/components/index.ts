export * from "./Footer";
export * from "./Logo";
export * from "./Button";
export * from "./Input";
export * from "./ThreeDotsIcon";
export * from "./ErrorBoundary";
