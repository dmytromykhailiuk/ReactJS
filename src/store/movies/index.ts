import * as MoviesAction from "./actions";
import * as MoviesSelector from "./selectors";

export { MoviesAction, MoviesSelector };
export * from "./effects";
export * from "./reducer";
