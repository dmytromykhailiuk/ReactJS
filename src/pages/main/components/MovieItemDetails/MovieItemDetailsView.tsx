import { Movie } from 'models/movie.model';
import React from 'react';
import classes from './MovieItemDetails.module.scss'
import { useDefaultImage } from 'shared/hooks';
import { Loader } from 'shared/components';

const loaderWrapperStyles = { marginTop: "80px", marginBottom: "80px" };

export interface MovieItemDetailsView {
  movieInOverview: Movie;
  isLoading: boolean;
}

const MovieItemDetailsView: React.FC<MovieItemDetailsView> = React.memo(({
  movieInOverview, 
  isLoading
}) => { 
  const [defaultImage, setError] = useDefaultImage(movieInOverview?.poster_path);

  return (
    <>
      { !isLoading ? (
        <div className={classes["movie-details"]}>
          <img 
            className={classes["movie-details__image"]} 
            src={ defaultImage || movieInOverview.poster_path }
            alt={movieInOverview.title}
            onError={setError}
          />
          <div className={classes["movie-details__info"]}>
            <div className={classes["movie-details__first-info-line"]}>
              <h2 className={classes["movie-details__title"]}>{ movieInOverview.title }</h2>
              <div className={classes["movie-details__rating"]}>{ movieInOverview.vote_average.toFixed(1) }</div>
            </div>
            <p className={classes["movie-details__category"]}>{ movieInOverview.genres.join(', ') }</p>
            <div className={classes["movie-details__third-info-line"]}>
              <div className={classes["movie-details__release-date"]}>{ movieInOverview.release_date.split('-')[0] }</div>
              { movieInOverview.runtime && <div>{ movieInOverview.runtime } min</div> }
            </div>
            <div className={classes["movie-details__overview"]}> { movieInOverview.overview } </div>
          </div>
        </div>
      ) : (
        <div style={loaderWrapperStyles}><Loader /></div>
      ) }
    </>
    
  )
})

export default MovieItemDetailsView;
