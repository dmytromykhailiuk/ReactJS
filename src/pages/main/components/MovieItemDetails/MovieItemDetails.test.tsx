import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { movies } from 'mocks/movies.mock';
import { Loader } from 'shared/components';
import MovieItemDetailsView from './MovieItemDetailsView';

describe('MovieItemDetails', () => {
  it('should match snepshot', () => {
    const wrapper = shallow(<MovieItemDetailsView movieInOverview={movies[0]} isLoading={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should show Lodaer when isLoading equal 'true'", () => {
    const wrapper = shallow(<MovieItemDetailsView movieInOverview={movies[0]} isLoading />);
    expect(wrapper.exists(Loader)).toBeTruthy();
  });
});
