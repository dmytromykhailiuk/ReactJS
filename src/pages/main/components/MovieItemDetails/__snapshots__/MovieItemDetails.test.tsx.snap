// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItemDetails should match snepshot 1`] = `
<Fragment>
  <div
    className="movie-details"
  >
    <img
      alt="Fifty Shades Freed"
      className="movie-details__image"
      onError={[Function]}
      src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
    />
    <div
      className="movie-details__info"
    >
      <div
        className="movie-details__first-info-line"
      >
        <h2
          className="movie-details__title"
        >
          Fifty Shades Freed
        </h2>
        <div
          className="movie-details__rating"
        >
          6.1
        </div>
      </div>
      <p
        className="movie-details__category"
      >
        Drama, Romance
      </p>
      <div
        className="movie-details__third-info-line"
      >
        <div
          className="movie-details__release-date"
        >
          2018
        </div>
        <div>
          106
           min
        </div>
      </div>
      <div
        className="movie-details__overview"
      >
         
        Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
         
      </div>
    </div>
  </div>
</Fragment>
`;
